### DISCUSSION TITLE
add query parameter support for GenericAPIView get_object method from lookup_field

### DISCUSSION THREAD

**Discussion by legitYosal** [encode/django-rest-framework#7789] [Ideas & Suggestions]
## Checklist

- [x] I have verified that that issue exists against the `master` branch of Django REST framework.
- [ ] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] This is not a usage question. (Those should be directed to the [discussion group](https://groups.google.com/forum/#!forum/django-rest-framework) instead.)
- [x] This cannot be dealt with as a third party library. (We prefer new functionality to be [in the form of third party libraries](https://www.django-rest-framework.org/community/third-party-packages/#about-third-party-packages) where possible.)
- [x] I have reduced the issue to the simplest possible case.
- [ ] I have included a failing test as a pull request. (If you are unable to do so we can still accept the issue.)
## Description
I wanted If I pass lookup_field in query parameters of URL my view accept it and return  retrieved object, and I funded a solution where I added query params to the kwargs of class as you know that get_object will search kwargs for lookup_field value, I want to know if I can make a pull request out of this??
## Simple Example
API example:
```
# views.py
class BookAPIView(GenericAPIView, RetrieveModelMixin):
    queryset = Book.objects.all()
    serializer_class = BookSerializer
    lookup_field = 'id'
    def get(self, request, *args, **kwargs):
        return self.retrieve(request, *args, **kwargs)
```
```
# urls.py
urlpatterns = [
    path('book/', BookAPIView.as_view(), name='bookApiView'), 
]
```
making `curl -X GET "http://localhost:8000/book/?id=12" -H "accept: application/json"` will return an error for cause of not passing `id` as a URL slug. 
## Solution
if we override genericAPIView like this:
```
# generics.py
class GenericAPIView(APIView):
    ...
    def dispatch(self, request, *args, **kwargs):
        lookup_param = request.GET.get(self.lookup_field, None)
        if lookup_param:
            kwargs[lookup_field] = lookup_param
        return Super(GenericAPIView, self).dispatch(request, *args, **kwargs)
    ...
```
now if lookup_field is in URL query parameters, the object will be retrieved.

**Comment by pietrotavares** [comment]
I missed this feature today as I had to split my ViewSet into two, less elegant, Views.
Also, this sounds like a somewhat common case.

Is there any chance for this PR?

**Reply by lovelydinosaur** [reply]
We're not going to support new features at this point, beyond keeping track with Django.

There's *plenty* of scope for extending REST framework either in your own codebase, or through third party packages.

**Reply by pietrotavares** [reply]
Sounds very reasonable to me.

People of the future:
I'll check in with the [DRF-extensions](https://github.com/chibisov/drf-extensions) folks, you should look there (in case I forget to edit this comment to reference the new feature).

In the meanwhile, I'll leave a reference to [this mixin from django-braces](https://django-braces.readthedocs.io/en/latest/other.html#orderablelistmixin) as it could potentially help others with a similar use case.

<br/>
@tomchristie Thanks for the quick reply, feel free to lock this discussion if you wish.
